### PR TITLE
fix: user defined data_dir

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

File: CHANGES-4.3.md [hunk 1]
```diff
@@ -16,6 +16,11 @@ File format:
 
 * For docker image, /opt/emqx/etc has been removed from the VOLUME list,
   this made it easier for the users to rebuild image on top with changed configs.
+* If `EMQX_NODE__DATA_DIR` override is in use, please note:
+  after this upgrade, the boot time generated files will be written to this
```
Comment (ieQu1) [inline CHANGES-4.3.md:20]:
```suggestion
  after this upgrade, the files generated at the boot time will be written to this
```

File: CHANGES-4.3.md [hunk 2]
```diff
@@ -16,6 +16,11 @@ File format:
...
+  override override dir instead of the default dir whichi is:
```
Comment (ieQu1) [inline CHANGES-4.3.md:21]: double "override"

File: CHANGES-4.3.md [hunk 3]
```diff
@@ -16,6 +16,11 @@ File format:
...
+  - `/opt/emqx/data` for docker
+  - `/var/lib/emqx` for deb/rpm installation
```
Comment (thalesmg) [inline CHANGES-4.3.md:23]:
```suggestion
  - `/var/lib/emqx` for deb/rpm installation
  - `$EMQX_DIR/data` if compiled from source, where `$EMQX_DIR` is the build output dir
```
?

**Review by thalesmg — DISMISSED** [review]

**Comment by zmstone** [comment]
Now I realise this is more complicated than I thought.
changing data_dir will only affect mnesia, in this sense, this PR is causing more inconsistent behaviour.

**Pull request closed without merging**